src/build_and_run_task.cr

```diff
@@ -0,0 +1,60 @@
+# Builds and runs a file (typically tasks.cr) with the passed in args
+class LuckyCli::BuildAndRunTask
+  private getter tasks_file : String
+  private getter args : String
+  private getter fake_stdout = IO::Memory.new
+  private getter fake_stderr = IO::Memory.new
+  private getter tasks_binary_path = "tmp/tasks_binary"
+
+  def initialize(@tasks_file, @args)
+  end
+
+  def self.call(*args, **named_args)
+    new(*args, **named_args).call
+  end
+
+  def call
+    FileUtils.mkdir_p("tmp")
+    build_status = build_tasks_binary
+    copy_fake_io_to_real_io
+
+    if build_status.success?
+      run_tasks_file
+    else
+      exit build_status.exit_code
+    end
+  end
+
+  private def build_tasks_binary
+    LuckyCli::Spinner.start("Compiling...") do
+      Process.run(
+        "crystal build #{tasks_file} -o #{tasks_binary_path} ",
+        shell: true,
+        input: STDIN,
+        output: fake_stdout,
+        error: fake_stderr
+      )
+    end
+  end
+
+  # STDOUT/STDERR needs to be printed after the tasks binary is built,
+  # otherwise the spinner will overwrite the output from building the binary
+  # (like if there is a compilation error when building).
+  #
+  # By creating "fake" io with IO::Memory we can collect the output and then
+  # print it after the build is finished.
+  private def copy_fake_io_to_real_io
+    STDOUT.print(fake_stdout.to_s)
+    STDERR.print(fake_stderr.to_s)
+  end
```

Comment:
This isn't ideal but works fine for our use case. Ideally the spinner would have some logic to print stdout/err above the spinner line, but that can be added later

Reply:
I think the process makes sense. Maybe we can call it `temp_stdout` or something? Using `fake` seems weird... but I guess we can just roll with it for now. We've been known to change up names of stuff later 😅

Reply:
Good idea! I like `temp_stdout/err`